src/api/java/com/minecolonies/api/entity/ai/statemachine/basestatemachine/BasicStateMachine.java

```diff
@@ -27,6 +27,11 @@
     @NotNull
     protected final Map<IStateEventType, ArrayList<T>> eventTransitionMap;
 
+    /**
+     * The current states list of transitions
+     */
+    protected ArrayList<T> currentStateTransitions;
+
```

Comment:
Specific type instead of `List<T>`

Reply:
hm there is a bunch of places in there where we're expecting that type though

Reply:
ye I can't change it anyway nicely

Reply:
Then refactor it.
You should never ever have a field with a concrete type. Especially ArrayList since it's api surface is identical to the normal list in all places that matter.

Reply:
What do you mean you can't change it anyway nicely?!?!??

Reply:
oh I think I just misunderstood, sure I'll change it to the list basetype :D